### PR TITLE
Ref/feature/prompt system

### PR DESCRIPTION
**Pull request by InamCZ**
#861mbh9uk

### CODE REVIEW

**Comment by MGH-Automator** [comment]
Task linked: [CU-861mqtrg5 Fix: Object jumps when parent changes](https://app.clickup.com/t/861mqtrg5)

File: Assets/Scripts/Visuals/PromptSystem.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        {
+            action?.Invoke(); 
+            data.window?.SetActive(false); 
+            data.taskCompleted.SetResult(this);
+        }
+    }
+
+    public class Yes : Action
+    {
+        public Yes(System.Action action)
```
Comment (cblech) [inline Assets/Scripts/Visuals/PromptSystem.cs:58]: Please add constructors with no parameters to all actions. That way, it's easier to just check the return of CreatePrompt

**Review by cblech — CHANGES_REQUESTED** [review]
When using CreateDialoge and leaving the notInWindowAction empty, the dialog does not open

**Review by cblech — CHANGES_REQUESTED** [review]
When a prompt is open, the background should gray out a bit, and the user should be prevented from clicking anything else than the prompt. 
If there is a `NotInWindow` Action defined, it should close when clicking next to the prompt, else it should do nothing

**Review by cblech — APPROVED** [review]

**Pull request merged**